### PR TITLE
Added "import numpy as np" statement to filter examples

### PR DESCRIPTION
**Pull request by edmundsj**
Noticed examples were missing a numpy import, and fail when copy-pasted and numpy is not already imported. Fixed all examples.

#### What does this implement/fix?
Some examples were failing in the scipy.signal.filters documentation

### CODE REVIEW

**Comment by ilayn** [comment]
Hi @edmundsj Thanks for the PR. Please have a look at #13049 for the discussion why this is the case

**Comment by edmundsj** [comment]
Thanks for the comment. Added my perspective on the thread.

File: scipy/signal/_filter_design.py [hunk 1]
```diff
@@ -1835,6 +1839,7 @@ def lp2hp(b, a, wo=1.0):
     --------
     >>> from scipy import signal
     >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
```
Comment (mdhaber) [inline scipy/signal/_filter_design.py:1842]:
```suggestion
```
Not used in the example.

**Review by mdhaber — APPROVED** [review]
Thanks @edmundsj!
In gh-13049 we decided to add these imports according to these rules - https://github.com/scipy/scipy/issues/13049#issuecomment-1199457608. The last rule is to leave existing imports alone, so I won't change all these (except removing one unnecessary one and fixing the PEP8 issue).

File: scipy/signal/_filter_design.py [hunk 2]
```diff
@@ -3131,6 +3137,8 @@ def cheby1(N, rp, Wn, btype='low', analog=False, output='ba', fs=None):
 
     >>> from scipy import signal
     >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+
```
Comment (mdhaber) [inline scipy/signal/_filter_design.py:3141]:
```suggestion
```

**Pull request merged**